crates/sui-core/src/histogram.rs

```diff
@@ -0,0 +1,402 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use futures::FutureExt;
+use parking_lot::Mutex;
+use prometheus::{
+    register_int_counter_vec_with_registry, register_int_gauge_vec_with_registry, IntCounterVec,
+    IntGaugeVec, Registry,
+};
+use std::collections::hash_map::DefaultHasher;
+use std::collections::{HashMap, HashSet};
+use std::hash::{Hash, Hasher};
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::runtime::Handle;
+use tokio::sync::mpsc;
+use tokio::sync::mpsc::error::TrySendError;
+use tokio::time::Instant;
+use tracing::error;
+
+type Point = u64;
+type HistogramMessage = (HistogramLabels, Point);
```

Comment:
Can we use an ID instead of replicating the labels with each message?  That makes each message very expensive ... typical labels for production Prom metrics can easily be 500 bytes - 1k.

Reply:
Labels is an Arc, so cloning is cheap

Reply:
Ah ok.